mabwiser/greedy.py

```diff
@@ -34,17 +34,17 @@ def partial_fit(self, decisions: np.ndarray, rewards: np.ndarray, contexts: np.n
 
     def predict(self, contexts: np.ndarray = None) -> Arm:
 
-        # Return a random arm with less than epsilon probability
-        if self.rng.rand() < self.epsilon:
-            return self.arms[self.rng.randint(0, len(self.arms))]
-
-        # Return the first arm with maximum expectation.
-        return argmax(self.arm_to_expectation)
+        # Return the first arm with maximum expectation
+        return argmax(self.predict_expectations())
 
     def predict_expectations(self, contexts: np.ndarray = None) -> Dict[Arm, Num]:
 
-        # Return a copy of expectations dictionary from arms (key) to expectations (values)
-        return self.arm_to_expectation.copy()
+        # Return a random expectation (between 0 and 1) for each arm with epsilon probability,
```

Comment:
we thought about in the initial/past design but did not want to have a predict_exp() function that returns different outputs if you call it twice.. 

It seems we are OK with this behavior now?

Reply:
We already has this behavior. Note that `predict_expectations` for [ThompsonSampling](https://github.com/fidelity/mabwiser/blob/master/mabwiser/thompson.py#L57) would return different outputs if it is called twice.

Reply:
I see, makes sense

Reply:
Re: determinism, I think there are ways to achieve it, such that consequent calls with the same `context` gives the same result (it would involve resetting the seed to _something_ each time `predict_expectations` is called). Not immediately necessary of course.